src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.CaptureInfoAnalyzer.cs

```diff
@@ -161,10 +161,8 @@ private void RecordSimpleGroupingCapture(RegexSimpleGroupingNode node, RegexOpti
                 // this grouping.  So if we note this grouping we'll end up not causing that error
                 // to happen, bringing out behavior out of sync with the native system.
                 var expr = node.Expression;
-                while (expr is RegexAlternationNode alternation)
-                {
-                    expr = alternation.Left;
-                }
+                if (expr is RegexAlternationNode alternation)
+                    expr = alternation.SequenceList[0];
 
```

Comment:
This must be something I'm missing. I don't get why those are equivalent. Per PR description:

> Our prior data representation for a|b|c|d|e was a deeply nested tree like so:
>
> ```
>           e
>          /
>      c  |  d
>     /
> a  |  b
> ```

If we have the node "e", left is going to be "c". But with the new implementation, `SequenceList[0]` will return "a".